ttyrec fails to run properly on Red Hat Enterprise Linux 5.x, giving errors like:

    Out of pty's
    /dev/ttyXX: No such file or directory


This patch allows it to build and run properly on RHEL (and likely CentOS):

    wget http://0xcc.net/ttyrec/ttyrec-1.0.8.tar.gz
    wget http://paperlined.org/apps/rhel/building/ttyrec-1.0.8.RHEL5.patch
    tar -xvzf ttyrec-1.0.8.tar.gz
    cd ttyrec-1.0.8
    patch -i ../ttyrec-1.0.8.RHEL5.patch

Source: http://paperlined.org/apps/rhel/building/ttyrec-1.0.8.RHEL5.txt

diff -rupN original/Makefile new/Makefile
--- original/Makefile	2006-06-11 10:52:50.000000000 -0500
+++ new/Makefile	2012-04-23 18:34:05.000000000 -0500
@@ -1,5 +1,5 @@
 CC = gcc
-CFLAGS = -O2
+CFLAGS = -O2 -DHAVE_openpty
 VERSION = 1.0.8
 
 TARGET = ttyrec ttyplay ttytime
@@ -10,7 +10,7 @@ DIST =	ttyrec.c ttyplay.c ttyrec.h io.c 
 all: $(TARGET)
 
 ttyrec: ttyrec.o io.o
-	$(CC) $(CFLAGS) -o ttyrec ttyrec.o io.o
+	$(CC) $(CFLAGS) -o ttyrec ttyrec.o io.o -lutil
 
 ttyplay: ttyplay.o io.o
 	$(CC) $(CFLAGS) -o ttyplay ttyplay.o io.o
diff -rupN original/ttyrec.c new/ttyrec.c
--- original/ttyrec.c	2006-06-11 10:52:50.000000000 -0500
+++ new/ttyrec.c	2012-04-23 18:26:41.000000000 -0500
@@ -71,7 +71,9 @@
 #define _(FOO) FOO
 
 #ifdef HAVE_openpty
-#include <libutil.h>
+/* #include <libutil.h> */
+#include <pty.h>  /* for openpty and forkpty */
+#include <utmp.h> /* for login_tty */
 #endif
 
 #if defined(SVR4) && !defined(CDEL)
